snow/vm.go

```diff
@@ -268,9 +293,40 @@ func (v *VM[I, O, A]) Initialize(
 		return err
 	}
 
+	// startAsyncAccepter with a detached context (do not assume context passed into VM initialize persists past the call)
+	v.startAsyncAccepter(context.WithoutCancel(ctx))
+
 	return nil
 }
 
+func (v *VM[I, O, A]) startAsyncAccepter(ctx context.Context) {
+	v.acceptedQueueWg.Add(1)
+
+	go v.log.RecoverAndPanic(func() {
+		defer func() {
+			// Mark the wg as done to ensure Shutdown does not block waiting for it
+			v.acceptedQueueWg.Done()
```

Comment:
this could hang in the following scenarios:
- let's say the acceptedQueue had 3 entries.
- The first one generate an error during processAccept.
- The current code would panic, and the anonymous defer function would get called.
- When `v.acceptedQueueWg.Done()` is called, it would never get satisfied, since no further `processAccept` calls are made.

Reply:
Nice catch! You're right this would hang in that scenario